norfair/tracker.py

```diff
@@ -228,38 +233,43 @@ def __init__(
             TrackedObject.initializing_count
         )  # Just for debugging
         TrackedObject.initializing_count += 1
-        self.setup_filter(initial_detection.points)
+        self.setup_filter(initial_detection.points, predictive_filter)
         self.detected_at_least_once_points = np.array([False] * self.num_points)
 
-    def setup_filter(self, initial_detection: np.array):
+    def setup_filter(
+        self, initial_detection: np.array, predictive_filter: KalmanFilter = None
+    ):
         initial_detection = validate_points(initial_detection)
 
-        dim_x = 2 * 2 * self.num_points  # We need to accomodate for velocities
         dim_z = 2 * self.num_points
         self.dim_z = dim_z
-        self.filter = KalmanFilter(dim_x=dim_x, dim_z=dim_z)
-
-        # State transition matrix (models physics): numpy.array()
-        self.filter.F = np.eye(dim_x)
-        dt = 1  # At each step we update pos with v * dt
-        for p in range(dim_z):
-            self.filter.F[p, p + dim_z] = dt
-
-        # Measurement function: numpy.array(dim_z, dim_x)
-        self.filter.H = np.eye(
-            dim_z,
-            dim_x,
-        )
+        if predictive_filter is not None:
+            self.filter = deepcopy(predictive_filter)
+        else:
+            dim_x = 2 * 2 * self.num_points  # We need to accommodate for velocities
+            self.filter = KalmanFilter(dim_x=dim_x, dim_z=dim_z)
+
+            # State transition matrix (models physics): numpy.array()
+            self.filter.F = np.eye(dim_x)
+            dt = 1  # At each step we update pos with v * dt
+
+            self.filter.F[:dim_z, dim_z:] = dt * np.eye(dim_z)
+
+            # Measurement function: numpy.array(dim_z, dim_x)
+            self.filter.H = np.eye(
+                dim_z,
+                dim_x,
+            )
 
-        # Measurement uncertainty (sensor noise): numpy.array(dim_z, dim_z)
-        # TODO: maybe we should open this one to the users, as it lets them
-        #       chose between giving more/less importance to the detections
-        self.filter.R *= 4.0
+            # Measurement uncertainty (sensor noise): numpy.array(dim_z, dim_z)
+            # TODO: maybe we should open this one to the users, as it lets them
```

Comment:
This TODO comment is off now.

Maybe we should actually change our approach, and make a function that provides Norfair's default instantiation of a `KalmanFilter` with the default parameters. Then, the user can modify this.

This would avoid a lot of code duplication if the user just wants to change one parameter of the filter. What do you think? @DonBraulio @aguscas